src/neo/IO/Caching/DataCache.cs

```diff
@@ -16,6 +16,7 @@ public class Trackable
         }
 
         private readonly Dictionary<TKey, Trackable> dictionary = new Dictionary<TKey, Trackable>();
+        private readonly HashSet<TKey> nonexistentKeySet = new HashSet<TKey>();
```

Comment:
It should have a maximum number of entries.

Reply:
Yes I also think so. I think we should have maximum number of entries both for this hashset andd dictionary above.

Reply:
`nonexistentKeySet` is a bloom filter.

Reply:
Yes, it could be a bloom filter.

Reply:
But there is a bloom filter in leveldb, we do we need another?

Reply:
We'll test the leveldb bloom filter.

Reply:
After testing it seems situation is much better after a 20-bit bloom filter added to levelDB startup. But still it will take somehow longer time for persisting than this commit, maybe due to more calling stacks

Reply:
Can we use https://github.com/neo-project/neo-modules/pull/253 ?